gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -336,17 +337,47 @@ var gpii = fluid.registerNamespace("gpii");
         var promiseSequence = fluid.promise.sequence(promises);
         // snapshots is an array of snapshotted settingshandler results (one for each lifecycle action block)
         promiseSequence.then(function (snapshots) {
-            if (saveSnapshot) {
-                var toSnapshot = fluid.copy(solutionRecord);
-                toSnapshot.settingsHandlers = {};
-                fluid.each(snapshots, function (snapshot) {
-                    if (snapshot !== undefined) { // can be removed once everything is settingsHandlers (GPII-1235)
-                        fluid.each(snapshot, function (handlerBlock, blockKey) {
-                            toSnapshot.settingsHandlers[blockKey] = handlerBlock;
-                        });
-                    }
-                });
+            var toSnapshot = fluid.copy(solutionRecord);
+            toSnapshot.settingsHandlers = {};
+            fluid.each(snapshots, function (snapshot) {
+                if (snapshot !== undefined) { // can be removed once everything is settingsHandlers (GPII-1235)
+                    fluid.each(snapshot, function (handlerBlock, blockKey) {
+                        toSnapshot.settingsHandlers[blockKey] = handlerBlock;
+                    });
+                }
+            });
+            if (fullSnapshot) {
                 sessionState.originalSettings[solutionId] = toSnapshot;
+            } else {
+                var shName, shBlock, setting;
+                // if we're doing an update, keep the settings that are already stored from the
+                // original snapshot, but augment it with any settings from the new snapshot
+                // that were not present in the original snapshot.
+                // First merge the basic structures:
+                var tmpOrigSettings = $.extend(true, {}, toSnapshot, sessionState.originalSettings[solutionId]);
+
+                // ensure that our augmented snapshot has any 'undefined' settings value copied over as well
+                // (since jQuerys extend will treat these as variables that are not set)
+                for (shName in toSnapshot.settingsHandlers) {
+                    shBlock = toSnapshot.settingsHandlers[shName];
+                    for (setting in shBlock.settings) { // will loop over undefined vals
+                        tmpOrigSettings.settingsHandlers[shName].settings[setting] = shBlock.settings[setting];
+                    }
+                }
+
+
+                // now recopy the original settings snapshot on top of the stored settings, but this time
+                // make sure that the settings from the original snapshot with a stored value of 'undefined' are copied over
+                // properly (again, jQuerys extend will treat these as variables that are not set)
+                if (sessionState.originalSettings[solutionId]) {
```

Comment:
It might be that there is no originalSettings[solutionId] at all, which would cause this to break on the attempt to access the settingsHandlers key of it.. or am I missing something?